src/core/types/DataTypes.sol

```diff
@@ -52,6 +50,58 @@ library Lockup {
         uint128 brokerFee;
     }
 
+    /// @notice Struct encapsulating the parameters of the `createWithDurations` functions.
+    /// @param sender The address distributing the assets, with the ability to cancel the stream. It doesn't have to be
+    /// the same as `msg.sender`.
+    /// @param recipient The address receiving the assets, as well as the NFT owner.
+    /// @param totalAmount The total amount, including the deposit and any broker fee, denoted in units of the asset's
+    /// decimals.
+    /// @param asset The contract address of the ERC-20 asset to be distributed.
+    /// @param cancelable Indicates if the stream is cancelable.
+    /// @param transferable Indicates if the stream NFT is transferable.
+    /// @param broker Struct encapsulating (i) the address of the broker assisting in creating the stream, and (ii) the
+    /// percentage fee paid to the broker from `totalAmount`, denoted as a fixed-point number. Both can be set to zero.
+    struct CreateWithDurations {
+        address sender;
+        address recipient;
+        uint128 totalAmount;
+        IERC20 asset;
+        bool cancelable;
+        bool transferable;
+        Broker broker;
+    }
+
+    /// @notice Struct encapsulating the parameters of the `createWithTimestamps` functions.
+    /// @param sender The address distributing the assets, with the ability to cancel the stream. It doesn't have to be
+    /// the same as `msg.sender`.
+    /// @param recipient The address receiving the assets, as well as the NFT owner.
+    /// @param totalAmount The total amount, including the deposit and any broker fee, denoted in units of the asset's
+    /// decimals.
+    /// @param asset The contract address of the ERC-20 asset to be distributed.
+    /// @param cancelable Indicates if the stream is cancelable.
+    /// @param transferable Indicates if the stream NFT is transferable.
+    /// @param timestamps Struct encapsulating (i) the stream's start time and (ii) end time, both as Unix timestamps.
+    /// @param broker Struct encapsulating (i) the address of the broker assisting in creating the stream, and (ii) the
+    /// percentage fee paid to the broker from `totalAmount`, denoted as a fixed-point number. Both can be set to zero.
+    struct CreateWithTimestamps {
+        address sender;
+        address recipient;
+        uint128 totalAmount;
+        IERC20 asset;
+        bool cancelable;
+        bool transferable;
+        Timestamps timestamps;
+        Broker broker;
+    }
+
+    /// @notice Enum representing the different distribution models used to create lockup streams.
+    /// @dev These distribution models determine the vesting function used in the calculations of the unlocked assets.
+    enum Model {
+        LOCKUP_LINEAR,
+        LOCKUP_DYNAMIC,
+        LOCKUP_TRANCHED
```

Comment:
Since this enum is a part of `Lockup` namespace, should we rename the values to:

```solidity
library Lockup {
    enum Model {
        LINEAR,
        DYNAMIC,
        TRANCHED
    }
}
```

Reply:
The return value won't have `LOCKUP` then.

Reply:
what do you mean?
the return value is a number (enums are `uint8`)

Reply:
The benefit is that for integrators, the return value can helps them distinguish between LOCKUP_LINEAR and some other LINEAR model, from the perspective of aggregators. That was the reason I prefixed them with `LOCKUP`

Reply:
> the return value is a number

Good point. Another thought I have is that on etherscan, if someone reads the function (or through RPC) and would see returned value as:
1. `LINEAR`
2. `LOCKUP_LINEAR`

Which one would be nice to read? I thought it would be nice to read `LOCKUP_LINEAR`, over `LINEAR`, what do you think?

Reply:
correct, for an integrators it looks better:

```solidity
import { Lockup } from "@sablier-labs/lockup/src/core/types/DataTypes.sol";

contract MyContract {
    // -- some logic ---
    function foo() public {
        Lockup.Model lockupModel = getLockupModel(id);
        if(lockupModel == LINEAR) {
            // do something
        }
        
        // vs
        if(lockupModel == LOCKUP_LINEAR) {
            // do something
        }
    }   
}
```